tests/ir/graph_view_test.py

```diff
@@ -0,0 +1,41 @@
+import pathlib
+import unittest
+
+import onnx
+
+from onnxscript import ir
+
+
+class GraphViewTest(unittest.TestCase):
+    def test_it_can_be_serialized_as_graph_proto(self):
+        data_path = (
+            pathlib.Path(__file__).parent.parent.parent
+            / "testdata/e2e_models/torchscript_model/torchscript_model.onnx"
+        )
+        model_proto = onnx.load(data_path)
+        model = ir.serde.deserialize_model(model_proto)
+
+        # Obtain a subset of nodes that belong to the first layer
+        nodes = [
+            node
+            for node in model.graph
+            if node.name is not None and node.name.startswith("/model/layers.0")
+        ]
+
+        inputs = set()
+        outputs = set()
+        for node in nodes:
+            inputs.update(node.inputs)
+            outputs.update(node.outputs)
+
+        graph_inputs = sorted(inputs - outputs, key=lambda val: val.name)
+        graph_outputs = sorted(outputs - inputs, key=lambda val: val.name)
+
+        graph_view = ir.GraphView(graph_inputs, graph_outputs, nodes=nodes)
+        model = ir.Model(graph_view, ir_version=8)
```

Comment:
can it support local function nodes?

Reply:
It can. We can create the same graphview on function nodes.

Reply:
How will the new `ir.Model` instance populate its 'functions' field?

Reply:
We can initialize a function using a graphview, or pass in functions from the original models, or operate directly on the graphview, depending on what we want to do.